cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -309,6 +313,31 @@ public void testMapreduceHistory() throws Exception {
   }
 
   /**
+   * Tests history of a non existing program
+   */
+  @Test
+  public void testNonExistingProgramHistory() throws Exception {
```

Comment:
Even you throw exception, the next test case would still get executed (that's how junit works). However, you can consider doing something like this.

```java
deploy(...);
int historyStatus = doPost(..).getStatusLine().getStatusCode();
int deleteStatus = doDelete(...).getStatusLine().getStatusCode();
Assert.assertTrue("Unexpected history status " + historyStatus + " and/or deleteStatus " + deleteStatus, 
                  historyStatus == 404 && deleteStatus == 200);
```